src/angular-app/bellows/apps/activity/activity-container.component.ts

```diff
@@ -42,10 +42,10 @@ class Activity {
   userHrefRelated: string;
   icon: string;
 
-  constructor(data: object = {}) {
+  constructor(data: Record<string, any> = {}) {
```

Comment:
if these are supposed to be keys of activity (due to cast below), then why don't we change `string` to `keyof Activity`?

Reply:
Typescript doesn't like that: `Type '{}' is missing the following properties from type 'Record<keyof Activity, any>': summary, action, content, changes, and 22 more.`

Reply:
oh right, the properties are optional. It would have to be `Partial<Record...>` then

Reply:
Trying `Partial<Record<keyof Activity, any>>` does work for that line, but Typescript still types the `for (const property of Object.keys(data))` has having properties of type `string`, rather than typing `property` as type `keyof Activity`. Which means that the `keyof Activity` typecases further down are still required. Going to leave this as-is since the further type refinements to `data` don't provide any benefit in practice.